internal/web.go

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2024 P0 Security, Inc
+// SPDX-License-Identifier: MPL-2.0
+
+package internal
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+)
+
+// Like http.Client, except:
+//   - handles authentication
+//   - converts to/from JSON
+//   - treats response codes 400 and higher as errors
+type P0ProviderData struct {
+	BaseUrl        string
+	Authentication string
+	Client         *http.Client
+}
+
+func (data *P0ProviderData) Do(req *http.Request, responseJson any) error {
```

Comment:
I would have expected responseJson to be a pointer `*any`, is this modifying the object you pass in for response json? 🤔